File: src/main/java/com/github/vitalliuss/cli/CommandLineArguments.java
package com.github.vitalliuss.cli;

import org.kohsuke.args4j.Option;

import java.net.URI;
import java.net.URISyntaxException;
import java.nio.file.Paths;

/**
 * Created by Vitali_Shulha on 16-Jun-16.
 */
public class CommandLineArguments {
    @Option(name="-i",aliases = {"--input"},usage="Directory to scan", required = true)
    private String input;

    @Option(name="-o",aliases = {"--output"},usage="Excel output file name",required = false)
    private String outputFileName = "output";

    public String getDirectoryToScan() {
        return input;
    }

    public String getOutputFileName() {
        return outputFileName;
    }

}

File: src/main/python/com/github/vitalliuss/cli/command_line_arguments.py
import argparse

class CommandLineArguments:
    def __init__(self):
        self.parser = argparse.ArgumentParser(description='Process command line arguments.')
        self.parser.add_argument('-i', '--input', dest='directory_to_scan', required=True, help='Directory to scan')
        self.parser.add_argument('-o', '--output', dest='output_file_name', default='output', help='Excel output file name')
        self.args = None

    def parse_arguments(self, args):
        self.args = self.parser.parse_args(args)
        return self.args

    @property
    def directory_to_scan(self):
        return self.args.directory_to_scan

    @property
    def output_file_name(self):
        return self.args.output_file_name